File: enhancement_worker_templates/lambda/js-layer/app/responses.js
module.exports = {
  testPassed: () => ({
    statusCode: 200,
    body: JSON.stringify({
      status: "ok",
    }),
  }),

  errorProcessing: (message) => ({
    statusCode: 500,
    body: message
      ? JSON.stringify({
          action: "error",
          content: message,
        })
      : undefined,
  }),
  rejectEvidence: () => ({
    statusCode: 406,
  }),
  badRequest: (message) => ({
    statusCode: 400,
    body: JSON.stringify({
      action: "error",
      content: message,
    }),
  }),
  notImplemented: () => ({
    statusCode: 501,
  }),
  processSuccess: (content) => ({
    statusCode: 200,
    body: JSON.stringify({
      action: "processed",
      content,
    }),
  }),
};

File: enhancement_workers/keyword-extractor-lambda/app/responses/responses.py
import json
from typing import Optional

def test_passed():
    return {
        'statusCode': 200,
        'body': json.dumps({
            'status': 'ok'
        })
    }

def process_success(content):
    return {
        'statusCode': 200,
        'body': json.dumps({
            'action': 'processed',
            'content': content
        })
    }

def reject_evidence():
    return {
        'statusCode': 406,
    }


def not_implemented():
    return {
        'statusCode': 501,
    }


def error_processing(message: Optional[str] = None):
    return _error_resp(500, message)


def bad_request(message: Optional[str] = None):
    return _error_resp(400, message)


def _error_resp(status_code: int, message: Optional[str] = None):
    rtn = {
        'statusCode': status_code,
    }

    if message is not None:
        rtn['body'] = json.dumps({
            'action': 'error',
            'content': message
        })

    return rtn
